syntax = "proto3";

import "serialization/proto/function_index.proto";

package serialization.proto;

message Function {
  uint32 parameters = 1;
  uint32 returns = 2;
  bytes content = 3;
}

message WrapperFunction {
  uint32 wrapped_module = 1;
  FunctionIndex wrapped_function = 2;
}

message FunctionTable {
  repeated Function functions = 1;
  repeated WrapperFunction wrappers = 2;
}
